plugins/node/opentelemetry-plugin-pg/src/pg.ts

```diff
@@ -78,8 +78,14 @@ export class PostgresPlugin extends BasePlugin<typeof pgTypes> {
               query,
               params
             );
+            if (plugin._config.applyCustomAttributesOnSpan) {
+              plugin._config.applyCustomAttributesOnSpan(span, query, params);
```

Comment:
would pass it as an object to avoid problems with checking params
```suggestion
              plugin._config.applyCustomAttributesOnSpan({ span, query, params });
```

Reply:
Just want to clarify real quick before I make this tweak.

You mean wrapping the arguments in an object for consumption in the function, perhaps something like:

```js
export interface PostgresCustomAttributesFunction {
  (ctx: QueryContext): void;
}

export interface QueryContext {
  span: Span;
  query: string | NormalizedQueryConfig;
  params?: unknown[];
}

export interface PostgresPluginConfig extends PluginConfig {
  applyCustomAttributesOnSpan?: PostgresCustomAttributesFunction;
}
```

I think that'd be a good way to avoid positional issues with the arguments. 

I'm also curious if it makes sense to split the text query and the `NormalizedQueryConfig` into separate fields. I reckon it would make for a more intuitive way to check what kind of query was being executed than the type checking that would need to be done in my most recent changes.

```js
export interface QueryContext {
  span: Span;
  query?: string;
  config?: NormalizedQueryConfig;
  params?: unknown[];
}
```

Reply:
yes 
```
export interface QueryContext {
  span: Span;
  query?: string;
  config?: NormalizedQueryConfig;
  params?: unknown[];
}
```
+1

Reply:
I'm not best with naming things but perhaps the name `applyCustomAttributesOnSpan` can be changed to something more meaningful now like `interceptOnQuery` etc. ?

Reply:
Perhaps the name `postQueryHook` would work? There's a bit of precedent because that's also a [function in the Http plugin](https://github.com/open-telemetry/opentelemetry-js/blob/5f4e479b2754e6dda2823999bb8419ea484ee9b2/packages/opentelemetry-plugin-http/src/types.ts#L77-L81)

Reply:
`postQueryHook` +1